public/js/charts/trendsCoverage.js

```diff
@@ -0,0 +1,173 @@
+// NOTE: requires('sheets');
+// NOTE: requires('aq');
+
+window.aq.charts.trendsCoverage = {};
+
+window.Chart.defaults.global.defaultFontSize = 10;
+
+const acquisition = document.getElementById('acquisition');
+
+async function handleLocationChange() {
+  const nodes = await window.aq.getNodes();
+  const countyCitiesMap = await window.sheets.getCountyCitiesMap();
+  function getAvg(sensor) {
+    const total = sensor.reduce((acc, c) => acc + c, 0);
+    return total / sensor.length;
+  }
+  // Format dates from nodes
+  function formatDate(date, langCode) {
+    const dateObj = new Date(date);
+    const day = dateObj.toLocaleString(langCode, { day: '2-digit' }); // DD
+    const month = dateObj.toLocaleString(langCode, { month: 'short' }); // MMM
+    return `${day} ${month}`;
+  }
+  // get chartlabels for the week in an array
+  const labels = [...Array(7)]
+    .map((_, i) => {
+      const d = new Date();
+      d.setDate(d.getDate() - i);
+      return d;
+    })
+    .map((item) => formatDate(item, 'en-US'))
+    .slice()
+    .sort((a, b) => new Date(a).getTime() - new Date(b).getTime());
```

Comment:
We can generate the dates in the order we want rather than having to sort e.g. if we want the descending, `d.getDate() - i` should do it, and if we want them ascending, `d.getDate() - 7 + i` should do it.